src/types/configFile.ts

```diff
@@ -0,0 +1,67 @@
+import { Type } from 'class-transformer'
+import {
+    IsObject,
+    IsOptional,
+    IsString,
+    registerDecorator,
+    ValidateNested,
+    ValidationOptions
+} from 'class-validator'
+
+function ValidateMatchPatterns(validationOptions?: ValidationOptions) {
+    return function(object: any, propertyName: string) {
+        registerDecorator({
+            name: 'ValidateMatchPatterns',
+            target: object.constructor,
+            propertyName,
+            options: {
+                message: `${propertyName} must be an object mapping file extensions to an array of Regex patterns`,
+                ...validationOptions
+            },
+            validator: {
+                validate(value: Record<string, unknown>) {
+                    for (const key in value) {
+                        if (!Array.isArray(value[key])) {
+                            return false
+                        }
+                    }
+
+                    return true
+                },
+            },
+        })
+    }
+}
+
+class CodeInsights {
+    @IsString({ each: true })
+    @IsOptional()
+    clientNames?: string[]
+
+    @IsOptional()
+    @IsObject()
```

Comment:
is the validation here sufficient? Any way I could break things with some invalid configuration?

Reply:
what if I provide a list instead, does that pass the IsObject validator?

Reply:
an array will fail IsObject, I've added a custom validator to verify that the values in the map are strings